enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,340 @@
+Theoretically, we could limit the number of user-trigerred gathering over time, but this problem can happen today when something is deleting the Insights operator pod on a regular basis with appropriate timing (so that data can be uploaded).  
+- When the newly spawned container is failing to upload the data then it can be running for some longer time (couple of hours). 
+
+
+## Design Details
+
+### Open Questions
+- ~~How to solve the problem with only one Insights analysis result available in a cluster? How to avoid the "uncertainty" of the Insights analysis results described in the [Risk and Mitigations](#risks-and-mitigations) section?~~ 
+  The gathering jobs will only check the status of the corresponding Insights analysis via the new API endpoint (See the section [Updates required to allow on-demand data gathering](#updates-required-to-allow-on-demand-data-gathering)) using the `insightsRequestID` attribute. 
+  The Insights analysis report will be downloaded by the operator when a gathering job is finished. The request ID of the Insights analysis will be compared to the corresponding job request ID and the job `AnalysisAvailable` condition will be updated based on this comparison. 
+- ~~Should the newly created gathering process update the clusteroperator conditions (e.g it fails to upload the data, should it update the `Degraded` status as it is today)? This can probably be racy too or it can lead to temporary and inconsistent conditions state.~~ 
+  Yes. The scenario is described in the [Updates required to allow on-demand data gathering](#updates-required-to-allow-on-demand-data-gathering) section.
+- ~~What all is required to spawn an extra container? Do we need a new image and a new binary (note that the Insights operator currently includes some additional features - such as pulling SCA certificates - that are not required for a new container/process)?~~ I think we can use the existing image and binary and introduce a new command.
+- ~~Is some clean-up of the old `datagather.insights.openshift.io` CRs (and completed containers/pods) required? I think it would make sense to keep them only for 24h.~~ The CRs older than 24 hours will be removed by the Insights operator. This time option will be configurable via the operator configuration. 
+
+### Test Plan
+
+This will require significant updates to the existing Insights operator testsuite. The tests will probably need to setup a persistent volume to easily consume the gathered archive. Following scenarios will need to be tested:
+
+- creation of a new `datagather.insights.openshift.io` CR triggers a new on-demand gathering
+- the status of the `datagather.insights.openshift.io` CR is updated correctly and propagated to the `insightsoperator.operator.openshift.io` CR
+- gathering requests are queued and there is only one at given time
+- error cases (related to the failing upload or download of the data) lead to same clusteroperator conditions as before
+- old jobs and CRs are pruned
```

Comment:
If there's a 24 hour delay on pruning, this sounds like it'll be rough to test, I assume for testing we will be able to configure the pruning period in some way that would be hard for a user to configure?

Reply:
Yes it should be configurable, but let's say not exposed by default (something similar as we do for the gathering interval right now. I will update it.

Reply:
I have expanded the wording here.